mapproxy/client/wms.py

```diff
@@ -199,22 +199,31 @@ def __init__(self, request_template, http_client=None):
 
     def get_legend(self, query):
         resp = self._retrieve(query)
-        format = split_mime_type(query.format)[1]
-        self._check_resp(resp)
-        return ImageSource(resp, image_opts=ImageOptions(format=format))
+        self._check_resp(resp, query.format)
+        if query.format == 'json':
+            return resp
+        else:
+            return ImageSource(resp, image_opts=ImageOptions(format=query.format))
```

Comment:
see comment above, you changed the input to `query.format` which affects other requests as well